### PR TITLE
scheduler: replace pauseLeader with two flags and add source filter to transferIn

### LINKED ISSUE #8621: scheduler: there is no source filter in transferIn

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #8621

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
Before this pr:
image
After this pr:
image

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/core/basic_cluster.go [hunk 1]
```diff
@@ -137,8 +137,10 @@ type StoreSetInformer interface {
 
 // StoreSetController is used to control stores' status.
 type StoreSetController interface {
-	PauseLeaderTransfer(id uint64) error
-	ResumeLeaderTransfer(id uint64)
+	PauseLeaderTransferIn(id uint64) error
```
Comment (rleungx) [inline pkg/core/basic_cluster.go:140]: How about adding a parameter?

File: pkg/schedule/schedulers/utils.go [hunk 2]
```diff
@@ -391,20 +391,38 @@ func (q *retryQuota) gc(keepStores []*core.StoreInfo) {
 	}
 }
 
-// pauseAndResumeLeaderTransfer checks the old and new store IDs, and pause or resume the leader transfer.
-func pauseAndResumeLeaderTransfer[T any](cluster *core.BasicCluster, old, new map[uint64]T) {
+// pauseAndResumeLeaderTransferIn checks the old and new store IDs, and pause or resume the leader transfer in.
+func pauseAndResumeLeaderTransferIn[T any](cluster *core.BasicCluster, old, new map[uint64]T) {
```
Comment (rleungx) [inline pkg/schedule/schedulers/utils.go:395]: ditto

File: pkg/core/basic_cluster.go (hunk 1 above)
Comment (lhy1024) [reply pkg/core/basic_cluster.go:140]: Is it something like `PauseLeaderTransferIn(id uint64, d direction) error`

Comment (rleungx) [reply pkg/core/basic_cluster.go:140]: yes, WDYT?

**Review by okJiang — APPROVED** [review]

**Comment by lhy1024** [comment]
PTAL @rleungx

**Review by rleungx — APPROVED** [review]

**Pull request merged**